### ISSUE TITLE
use-xdg-base-directories does not effect $PATH

### ISSUE THREAD

**Issue by bestlem** [LnL7/nix-darwin#943]
I have set nix.settings.use-xdg-base-directories = true

This means that running `nix  build .#"homeConfigurations.user@machine.activationPackage"` and running the activation home-manager puts things in $XDG_STATE_HOME/nix/profile rather than ~/.nix_profile according to https://nixos.org/manual/nix/stable/command-ref/conf-file

However the path set by nix-darwin includes ~/.nix-profile.bin rather than the XDG one.

Am I correct is assuming that |/.nix-profile should not exist and that the $PATH should include $XDG_STATE_HOME/nix/profile/bin ?

I suspect others are not noticing this as they always use darwin-rebuild with home-manager.useUserPackages = true; so the path is /etc/profiles/per-user/<user>/bin  which is on the PATH . 

I was trying not to have to use sudo each time so I used the build above - should I not be doing that?

**Comment by pwnwriter** [comment]
I'm having the same issue, did you figure out any solution?

**Comment by sellout** [comment]
I think this and #947 are effectively the same issue. #962 was attempting to address it, but was abandoned in the face of no fully-satisfactory solution. I think what’s there is at least a step toward correctness and I have adopted a similar change locally.

**Comment by pwnwriter** [comment]
> I have adopted a similar change locally.

Could you elaborate on this a bit?

**Comment by sellout** [comment]
> Could you elaborate on this a bit?

Yeah, sorry – it’s in that PR, but buried a bit: https://github.com/LnL7/nix-darwin/pull/962#discussion_r1763516601

In https://github.com/LnL7/nix-darwin/pull/962#issuecomment-2356377603 I also said
>  I’m not seeing /nix/profile in my PATH or NIX_PROFILES, so it seems like the XDG_STATE_HOME is set sufficiently early somehow.

But I was wrong about that – after cleaning up some pre-Nix config, I was seeing the same issue with `XDG_STATE_HOME` being set too late. So, this only works if you’re using the default value for `XDG_STATE_HOME`, since that’s handled by the second entry.

The real solution is likely deeper – so that the correct values end up in `PATH`, the PAM config, etc.

**Comment by smancill** [comment]
I just found that a workaround is to add the XDG directory to `environment.profiles` with the proper order:

``` nix
environment.profiles = lib.mkOrder 700 [ "\$HOME/.local/state/nix/profile" ];
```

This will make sure it is added to all variables that depend on `environment.profiles` in the proper order:

```
export PATH=$HOME/.local/state/nix/profile/bin:$HOME/.nix-profile/bin:/etc/profiles/per-user/$USER/bin:/run/current-system/sw/bin:/nix/var/nix/profiles/default/bin:/usr/local/bin:/usr/bin:/usr/sbin:/bin:/sbin
export TERMINFO_DIRS="$HOME/.local/state/nix/profile/share/terminfo:$HOME/.nix-profile/share/terminfo:/etc/profiles/per-user/$USER/share/terminfo:/run/current-system/sw/share/terminfo:/nix/var/nix/profiles/default/share/terminfo:/usr/share/terminfo"
export XDG_CONFIG_DIRS="$HOME/.local/state/nix/profile/etc/xdg:$HOME/.nix-profile/etc/xdg:/etc/profiles/per-user/$USER/etc/xdg:/run/current-system/sw/etc/xdg:/nix/var/nix/profiles/default/etc/xdg"
export XDG_DATA_DIRS="$HOME/.local/state/nix/profile/share:$HOME/.nix-profile/share:/etc/profiles/per-user/$USER/share:/run/current-system/sw/share:/nix/var/nix/profiles/default/share"

export NIX_PROFILES="/nix/var/nix/profiles/default /run/current-system/sw /etc/profiles/per-user/$USER $HOME/.nix-profile $HOME/.local/state/nix/profile"
```

**Comment by bestlem** [comment]
Again the issue is that this is for the whole system.

Some users might have XDG and others not, Others might have things set so that XDG_CACHE is ~/Library/Caches,

Basically XDG is a user setting and not a system one.
